### PR TITLE
Create user doc if one doesn't exist for a user

### PR DESCRIPTION
**Pull request by CarsonHerness**
Apologies in advance for how long this ended up being. Here are the major changes:

- If an authenticated user doesn't have a user doc, one is created
- Users now have the option to add or update their name
- The interface of the "Manage Account" page is improved
- Minor bugs with translation were also fixed
- eslint dependencies were moved to dev dependencies, and for some reason it also alphabetized the dependencies

### CODE REVIEW

File: src/components/Admin/Authentication/ChangeName.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+              <Flex alignItems="center" justifyContent="center" marginTop="1em">
+                <CheckCircleIcon color="green.500" />
+                <Text fontSize="lg">{t('nameModal.success')}</Text>
+              </Flex>
+            ) : (
+              <ModalBody>
+                <Box marginY={1}>
+                  {passwordUser && (
+                    <PasswordFormInput
+                      label={'password'}
```
Comment (acgrobman) [inline src/components/Admin/Authentication/ChangeName.tsx:165]: This should be translated, appears directly on website

File: public/locales/es/common.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
     "close": "Cierra",
+    "submit": "Enviar",
```
Comment (acgrobman) [inline public/locales/es/common.json:3]: Verb format conjugated?

File: src/components/Admin/Authentication/ChangeName.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+          .then(() => {
+            // Update Firestore user document
+            firestore
+              .collection('users')
+              .doc(user.uid)
+              .update({
+                name: newName,
+              })
+              .then(() => setNameChangeComplete(true))
+              .catch(error => {
```
Comment (acgrobman) [inline src/components/Admin/Authentication/ChangeName.tsx:107]: Does firebase require us to catch and rethrow vs. never catching to begin with?

File: public/locales/es/common.json (hunk 2 above)
Comment (CarsonHerness) [reply public/locales/es/common.json:3]: The internet says "Enviar" is the conventional form, here's one such [reference of UX Spanish translations](https://ux.stackexchange.com/questions/80833/reference-for-spanish-ui)

File: src/components/Admin/Authentication/ChangeName.tsx (hunk 3 above)
Comment (CarsonHerness) [reply src/components/Admin/Authentication/ChangeName.tsx:107]: Good point, I can just remove the catch statement ([this Stack Overflow](https://stackoverflow.com/questions/42167274/rethrowing-error-in-promise-catch) post explains it well)

File: src/contexts/AuthContext.tsx [hunk 4]
```diff
@@ -30,50 +35,88 @@ const AuthProvider: React.FC<Props> = ({children}: Props) => {
...
+      // updates their account information, this change is reflected on the user's
+      // account page.
+      const unsubscribe = firestore
+        .collection('users')
+        .doc(user.uid)
+        .onSnapshot(snapshot => {
+          if (snapshot.exists) {
+            const userData = snapshot.data();
+            if (userData) {
+              if (userData.admin !== undefined) setIsAdmin(userData.admin);
```
Comment (akampen) [inline src/contexts/AuthContext.tsx:83]: should we also check that userData.admin is a boolean to avoid throwing an error?

**Review by akampen — APPROVED** [review]

File: src/contexts/AuthContext.tsx (hunk 4 above)
Comment (CarsonHerness) [reply src/contexts/AuthContext.tsx:83]: Good point, I'll add some type safety stuff here

File: public/locales/es/administration.json [hunk 5]
```diff
@@ -10,30 +10,38 @@
     "accountDisabled": "La cuenta ha sido desactivada. Si crees que es un error, contacta a un administrador",
     "incorrectPassword": "Contraseña incorrecta. Vuelve a intentar o vuelve a ponerla. También es posible que creaste tu cuenta con otro metódo",
     "accountExists":"Ya existe una cuenta con el mismo correo electrónico. Por favor, inicia sesión o usa un correo electrónico diferente",
-    "modalLaunch": {
+    "passwordModalLaunch": {
         "change": "Actualiza contraseña",
```
Comment (hillaryrguez) [inline public/locales/es/administration.json:14]: Actualiza la contraseña

File: public/locales/es/administration.json [hunk 6]
```diff
@@ -10,30 +10,38 @@
...
         "forgot": {
             "text": "¿Has olvidado tu contraseña? ",
             "link": "Vuelve a ponerla"
         }
     },
-    "modalHeader": {
+    "passwordModalHeader": {
         "change":"Actualiza contraseña",
```
Comment (hillaryrguez) [inline public/locales/es/administration.json:21]: Actualiza la contraseña

File: public/locales/es/administration.json [hunk 7]
```diff
@@ -10,30 +10,38 @@
...
     }, 
     "currentPassword": "Contraseña actual",
     "newPassword": "Contraseña nueva",
     "confirmNewPassword": "Confirma la contraseña nueva",
     "password": "Contraseña",
+    "passwordHelpMessage": "Introduce tu contraseña actual antes de actualizar tu cuenta",
     "confirmPassword": "Confirma la contraseña",
-    "submitLabel": {
+    "passwordSubmitLabel": {
         "change": "Actualiza contraseña",
```
Comment (hillaryrguez) [inline public/locales/es/administration.json:35]: Actualiza la contraseña

File: public/locales/es/administration.json [hunk 8]
```diff
@@ -10,30 +10,38 @@
...
         "forgot": "Envia el correo electrónico"
     },
+    "nameModal": {
+        "launchButton": "Actualiza tu nombre",
+        "header": "Actualiza nombre",
+        "submitButton": "Actualiza nombre",
```
Comment (hillaryrguez) [inline public/locales/es/administration.json:41]: Actualiza tu nombre

**Pull request merged**